service/controller/v1/resource/app/desired.go

```diff
@@ -0,0 +1,80 @@
+package app
+
+import (
+	"context"
+
+	applicationv1 "github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	corev1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/release-operator/service/controller/v1/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	releaseCR, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var appCRs []*applicationv1.App
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "computing desired state")
+
+		for _, authority := range releaseCR.Spec.Authorities {
+			appCR, err := r.newAppCR(ctx, releaseCR, authority)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			appCRs = append(appCRs, appCR)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "computed desired state")
+	}
+
+	return appCRs, nil
+}
+
+func (r *Resource) newAppCR(ctx context.Context, releaseCR corev1.Release, authority corev1.ReleaseSpecAuthority) (*applicationv1.App, error) {
+	appCR := &applicationv1.App{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "App",
+			APIVersion: "core.giantswarm.io",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: authority.HelmReleaseName(),
+			Labels: map[string]string{
+				key.LabelApp:            authority.Name,
+				key.LabelManagedBy:      key.ProjectName,
+				key.LabelOrganization:   key.OrganizationName,
+				key.LabelReleaseVersion: key.ReleaseVersion(releaseCR),
+				key.LabelServiceType:    key.ServiceTypeManaged,
+			},
```

Comment:
We'll need to do something for internal versioning for app CRs but its not agreed yet. At the moment chart-operator uses an annotation `giantswarm.io/version-bundle` but its likely that will change.

@TheoBrigitte I think its best we leave it like this for now. I'll let you know once this is agreed in the App Catalog spec. Sorry about the hassle.

Reply:
Nothing was agreed at the time I wrote this code.

But now we have a new `OPERATOR.giantswarm.io/version` label in fmt https://github.com/giantswarm/fmt/blob/master/kubernetes/annotations_and_labels.md#common-labels

Here we could have something like: `app-operator.giantswarm.io/version=1.0.0`

Reply:
@TheoBrigitte Yes we discussed this in Batman planning and we'll use the label `app-operator.giantswarm.io/version=1.0.0`.

I still need to get the spec PR merged. But we might as well get it sorted now rather than delay.

Reply:
We also have an ongoing discussion in Magic in order to know where we get the app-operator version from, and how its managed through updated.

So I added the `app-operator.giantswarm.io/version` label with no value for now. We will sort the value in a different PR.